scene/main/node.cpp

```diff
@@ -1727,6 +1727,52 @@ TypedArray<Node> Node::find_children(const String &p_pattern, const String &p_ty
 	return ret;
 }
 
+bool Node::is_child_of_exposed_node(const Node *p_owner) const {
+	HashSet<const Node *> visited;
+	if (p_owner) {
+		const Node *n = p_owner->get_parent();
+		while (n) {
+			visited.insert(n);
+			n = n->data.parent;
+		}
+	}
+	const Node *common_parent = this;
+	common_parent = common_parent->data.parent;
```

Comment:
```suggestion
	const Node *common_parent = data.parent;
```
Can do this in one step